pkg/datapath/loader/base.go

```diff
@@ -239,14 +240,29 @@ func (l *Loader) reinitializeXDPLocked(ctx context.Context, extraCArgs []string)
 	if option.Config.XDPMode == option.XDPModeDisabled {
 		return nil
 	}
+
 	for _, dev := range option.Config.GetDevices() {
+
 		// When WG & encrypt-node are on, the devices include cilium_wg0 to attach bpf_host
 		// so that NodePort's rev-{S,D}NAT translations happens for a reply from the remote node.
 		// So We need to exclude cilium_wg0 not to attach the XDP program when XDP acceleration
 		// is enabled, otherwise we will get "operation not supported" error.
 		if dev == wgTypes.IfaceName {
 			continue
 		}
+
+		// Don't enable XDP on links that match the option.Config.XDPIgnoreDeviceNameRegex regex
+		// This is useful for when you want XDP but a required device driver on the system does
+		// not yet have support for XDP in the kernel.
+		if len(option.Config.XDPIgnoreDeviceNameRegex) > 0 {
+
+			ignored, _ := regexp.Compile(option.Config.XDPIgnoreDeviceNameRegex)
```

Comment:
Nit: call `regexp.MustCompile` and assign to an (unexported) package-scoped variable. Compiling regular expressions is expensive, no need to do that repeatedly.